extensions/src/org/pentaho/platform/plugin/services/security/userrole/ldap/DefaultLdapUserRoleListService.java

```diff
@@ -114,6 +117,13 @@ public void afterPropertiesSet() throws Exception {
         roles.add( roleString );
       }
     }
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
+      if ( !roles.contains( extraRole ) ) {
```

Comment:
Once again, in an LDAP case, this is likely to be poorly performing. With 1000 roles from LDAP, and 5 "extra roles", this could take a considerable time in the "lookup" part. If there are only ever 2 extra roles, then it's likely OK - it depends on what the "extra roles" are.